cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/model/Group.java

```diff
@@ -32,24 +34,32 @@
 
     private final Optional<CloudFileSystemView> identity;
 
+    private final List<CloudInstance> deletedInstances;
+
     public Group(String name, InstanceGroupType type, Collection<CloudInstance> instances, Security security, CloudInstance skeleton,
             InstanceAuthentication instanceAuthentication, String loginUserName, String publicKey,
             int rootVolumeSize, Optional<CloudFileSystemView> identity) {
-        this.name = name;
-        this.type = type;
-        this.instances = ImmutableList.copyOf(instances);
-        this.security = security;
-        this.skeleton = Optional.ofNullable(skeleton);
-        this.instanceAuthentication = instanceAuthentication;
-        this.publicKey = publicKey;
-        this.loginUserName = loginUserName;
-        this.rootVolumeSize = rootVolumeSize;
-        this.identity = identity;
+        this(name, type, instances, security, skeleton, new HashMap<>(), instanceAuthentication, loginUserName, publicKey,
+            rootVolumeSize, identity, new ArrayList<>());
+    }
+
+    public Group(String name, InstanceGroupType type, Collection<CloudInstance> instances, Security security, CloudInstance skeleton,
+            InstanceAuthentication instanceAuthentication, String loginUserName, String publicKey,
+            int rootVolumeSize, Optional<CloudFileSystemView> identity, List<CloudInstance> deletedInstances) {
+        this(name, type, instances, security, skeleton, new HashMap<>(), instanceAuthentication, loginUserName, publicKey,
+            rootVolumeSize, identity, deletedInstances);
     }
 
     public Group(String name, InstanceGroupType type, Collection<CloudInstance> instances, Security security, CloudInstance skeleton,
             Map<String, Object> parameters, InstanceAuthentication instanceAuthentication, String loginUserName,
             String publicKey, int rootVolumeSize, Optional<CloudFileSystemView> identity) {
+        this(name, type, instances, security, skeleton, parameters, instanceAuthentication, loginUserName, publicKey,
+            rootVolumeSize, identity, new ArrayList<>());
+    }
+
+    public Group(String name, InstanceGroupType type, Collection<CloudInstance> instances, Security security, CloudInstance skeleton,
```

Comment:
**suggestion:** The number of arguments and overloadings of this constructor makes me think it would be good to associate it with a Builder.

Reply:
I don't disagree, but these constructors are called _so many_ places. Changing this to a builder would dwarf the actual changes to fix the problem. That probably needs to be its own commit.